delft/sequenceLabelling/models.py

```diff
@@ -23,144 +22,29 @@
-        preprocessor.return_lengths = True
-        config.use_crf = True
-        config.use_chain_crf = True
-        return BidLSTM_ChainCRF_FEATURES(config, ntags)
-
-    elif config.architecture == BidLSTM_CRF_CASING.name:
-        preprocessor.return_word_embeddings = True
-        preprocessor.return_casing = True
-        preprocessor.return_chars = True
-        preprocessor.return_lengths = True
-        config.use_crf = True
-        return BidLSTM_CRF_CASING(config, ntags)
-
-    elif config.architecture == BERT.name:
-        preprocessor.return_bert_embeddings = True
-        config.labels = preprocessor.vocab_tag
-        return BERT(config, 
-                    ntags, 
-                    load_pretrained_weights=load_pretrained_weights, 
-                    local_path=local_path)
-
-    elif config.architecture == BERT_CRF.name:
-        preprocessor.return_bert_embeddings = True
-        config.use_crf = True
-        config.labels = preprocessor.vocab_tag
-        return BERT_CRF(config, 
-                        ntags, 
-                        load_pretrained_weights=load_pretrained_weights, 
-                        local_path=local_path)
-
-    elif config.architecture == BERT_ChainCRF.name:
-        preprocessor.return_bert_embeddings = True
-        config.use_crf = True
-        config.use_chain_crf = True
-        config.labels = preprocessor.vocab_tag
-        return BERT_ChainCRF(config, 
-                        ntags, 
-                        load_pretrained_weights=load_pretrained_weights, 
-                        local_path=local_path)
-
-    elif config.architecture == BERT_CRF_FEATURES.name:
-        preprocessor.return_bert_embeddings = True
-        preprocessor.return_features = True
-        config.use_crf = True
-        config.labels = preprocessor.vocab_tag
-        return BERT_CRF_FEATURES(config, 
-                                ntags, 
-                                load_pretrained_weights=load_pretrained_weights, 
-                                local_path=local_path)
+    def get_transformer_config(self):
+        # transformer config (PretrainedConfig) if a pretrained transformer is used in the model
+        return None
 
-    elif config.architecture == BERT_CRF_CHAR.name:
-        preprocessor.return_bert_embeddings = True
-        preprocessor.return_chars = True
-        config.use_crf = True
-        config.labels = preprocessor.vocab_tag
-        return BERT_CRF_CHAR(config, 
-                            ntags,      
-                            load_pretrained_weights=load_pretrained_weights, 
-                            local_path=local_path)
+    def init_transformer(self, config, transformer: Transformer, load_pretrained_weights, local_path):
+        if config.transformer_name is not None and transformer is None:
+            name = config.transformer_name
+            transformer = Transformer(name, config_file=config.transformer_name, delft_local_path=local_path)
```

Comment:
So here why not simply always init a transformer object without carrying `transformer` as extra parameter from `wrapper.py` to here?